tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestAssemblyInfoRunAssemblyInitializeIntegration.cs

```diff
@@ -49,17 +49,27 @@ internal static CallTargetState OnMethodBegin<TTarget, TContext>(TTarget instanc
             return CallTargetState.GetDefault();
         }
 
-        if (!testContext.TryDuckCast<TestContextStruct>(out var context))
+        lock (instance.Instance!)
         {
-            return CallTargetState.GetDefault();
+            instance.AssemblyCleanupMethod ??= EmptyCleanUpMethodInfo;
         }
 
-        lock (instance.Instance!)
+        string? assemblyName;
+        if (testContext.TryDuckCast<TestContextStruct>(out var context) && context.TestMethod is { AssemblyName: { } contextAssemblyName })
```

Comment:
nit: should the second part of the `if` be _inside_ the outer one?

i.e. is there ever a case where we _can_ duck cast as `TestContextStruct`, but assembly is null, and so we should _also_ try to duck cast `ITestAssemblyInfoWithAssembly`? Or is it either/or with the duck casting?

Reply:
changed in https://github.com/DataDog/dd-trace-dotnet/pull/6686/commits/a565f8452c22f2fabbeaf9e33fad85bd090faa55